### PR TITLE
generate istio.io related manifest when ISTIO_INJECT=true

### PR DESCRIPTION
**Pull request by carolynhu**

### CODE REVIEW

**Comment by mandarjog** [comment]
Hey Carolyn can you add a blurb about why it is being refactored ? What do you want to achieve.

**Comment by carolynhu** [comment]
@mandarjog https://github.com/istio/istio/issues/19001

**Comment by carolynhu** [comment]
a sample generated yaml
[twopods.yaml.zip](https://github.com/istio/tools/files/3860725/twopods.yaml.zip)

**Comment by mandarjog** [comment]
I think this is fine, @richardwxn any objections?

File: perf/benchmark/templates/fortio.yaml [hunk 1]
```diff
@@ -206,13 +209,15 @@ spec:
...
 {{- end }}
+{{- end }}
 
 {{- $fortioserver := dict "name" "fortioserver" "Values" .Values "V" .Values.server}}
 {{- template "fortio" $fortioserver }}
 {{- $fortioclient := dict "name" "fortioclient" "Values" .Values "V" .Values.client }}
 {{- template "fortio" $fortioclient }}
 
 ---
+{{- if $.Values.server.inject }}
```
Comment (richardwxn) [inline perf/benchmark/templates/fortio.yaml]: this should be Values.server.inject or Values.client.inject?

Comment (carolynhu) [reply perf/benchmark/templates/fortio.yaml]: either is ok

**Pull request merged**